django/db/models/fields/__init__.py

```diff
@@ -857,10 +863,39 @@ def _get_default(self):
                 return self.default
             return lambda: self.default
 
+        if self.has_db_default():
+            return lambda: DB_DEFAULT
+
         if not self.empty_strings_allowed or self.null and not connection.features.interprets_empty_strings_as_nulls:
             return return_None
         return str  # return empty string
 
+    def has_db_default(self):
+        """Return a boolean of whether this field has a database default value."""
+        return self.db_default is not NOT_PROVIDED
+
+    def db_default_sql(self, model, schema_editor):
+        from django.db.models.expressions import Expression, Value
+        from django.db.models.sql.query import Query
+
+        if isinstance(self.db_default, Expression):
+            db_default = self.db_default
+        else:
+            db_default = Value(self.db_default)
+
+        query = Query(model=model)
+        db_default = db_default.resolve_expression(query)
+        connection = schema_editor.connection
+        compiler = query.get_compiler(connection=connection)
+        sql, params = compiler.compile(db_default)
+        if connection.vendor == 'sqlite':
+            sql %= tuple(
+                f'"{value}"' if isinstance(value, str) else value
```

Comment:
Can `schema_editor.quote_name` be used here instead?
```suggestion
                schema_editor.quote_name(value)
```

Reply:
It would work but it would be a kind of implementation detail abuse since `value` is not an SQL identifier.

This whole method should likely live on `connection.ops` instead.

Reply:
I used `quote_name` in an earlier draft and changed it to this for the same reason. I'll probably move the method to `connection.ops` later.

Reply:
👍

Reply:
I discovered `schema_editor.quote_value` exists, so I've used that instead.